proto/svc-storage-grpc.proto

```diff
@@ -36,161 +23,14 @@ message ReadyResponse {
-    uint32 vertiport_id_departure = 6;
-    //pad_id_departure
-    uint32 pad_id_departure = 7;
-    //vertiport_id_destination
-    uint32 vertiport_id_destination = 8;
-    //pad_id_destination
-    uint32 pad_id_destination = 9;
-    //estimated_departure
-    google.protobuf.Timestamp estimated_departure = 10;
-    //estimated_arrival
-    google.protobuf.Timestamp estimated_arrival = 11;
-    //actual_departure
-    optional google.protobuf.Timestamp actual_departure = 12;
-    //actual_arrival
-    optional google.protobuf.Timestamp actual_arrival = 13;
-    //flight_release_approval
-    optional google.protobuf.Timestamp flight_release_approval = 14;
-    //flight_plan_submitted
-    optional google.protobuf.Timestamp flight_plan_submitted = 15;*/
-    //flightStatus
-    FlightStatus flightStatus = 16;
-    //flightPriority
-    //FlightPriority flightPriority = 17;
-
-}
-
-//FlightPlans
-message FlightPlans {
-    //array/vector of flight items
-    repeated FlightPlan flightPlans = 1;
-}
-
-//FlightPlans
-message FlightPlanFilter {
-    //todo add filter parameters
-}
-
-
-//Aircraft
-message Aircraft {
-    //id
-    uint32 id = 1;
-    //string make = 2;
-    //string model = 3;
-    string nickname = 4;
-    //uint32 passenger_capacity = 5;
-    //google.protobuf.Timestamp end_lifespan = 6;
-    //google.protobuf.Timestamp next_maintenance = 7;
-    //google.protobuf.Timestamp last_maintenance = 8;
-    //google.protobuf.Timestamp last_location_update = 9;
-    //float last_location_latitude = 10;
-    //float last_location_longitude = 11;
-    //float voltage_x = 12;
-    //float voltage_y = 13;
-    //float amperage_x = 14;
-    //float amperage_y = 15;
-}
-//Aircrafts
-message Aircrafts {
-    //array/vector of flight items
-    repeated Aircraft aircrafts = 1;
-}
-//AircraftFilter
-message AircraftFilter {
-    //todo add filter parameters
-}
-//Vertiport
-message Vertiport {
-    uint32 id = 1;
-    string label = 2;
-    float latitude = 3;
-    float longitude = 4;
-    //repeated uint32 engineers = 5;
-    repeated uint32 pads = 6;
-    //uint32 elevation = 7;
-}
-//Vertiports
-message Vertiports {
-    repeated Vertiport vertiports = 1;
-}
-//VertiportFilter
-message VertiportFilter {
-    //todo add filter parameters
-}
-//Pad
-message Pad {
-    uint32 id = 1;
-    //bool enabled = 2;
-    //bool charging_enabled = 3;
-    //float charging_rate_kw = 4;
-    //string restrictions = 5;
-    float latitude = 6;
-    float longitude = 7;
-    bool parked = 8;
-}
-//Pads
-message Pads {
-    repeated Pad pads = 1;
-}
-//PadFilter
-message PadFilter {
-    //todo add filter parameters
-}
-
-//Pilot
-message Pilot {
-    uint32 id = 1;
-    string first_name = 2;
-    string last_name = 3;
-    /*string wallet_address = 4;
-    string type = 5;*/
-}
-//Pilots
-message Pilots {
-    repeated Pilot pilots = 1;
-}
-//PilotFilter
-message PilotFilter {
-    //todo add filter parameters
+//SearchFilter
+message SearchFilter {
```

Comment:
I like the idea of the generic filter.
But to make it useful for more complex queries we would need to add more operators like BETWEEN [date1, date2], IN [id1, id2, id3,...], NOT IN, IS (NOT) NULL,...
and also add capability to add multiple filters with AND and OR.

For inspiration we can look at how Hasura implements `where` filters as JSON: https://hasura.io/docs/latest/queries/postgres/query-filters/ ( Ignore the grapqhl part, just look at `where` which is pure json)

For scheduler R1 we need at least BETWEEN dates  and adding multiple conditions with AND. But if time's too short I can get all data and filter on scheduler end.

Reply:
Will definately check it out to improve the filters!